### PR TITLE
Update: add fixer for no-floating-decimal (fixes #7070)

### LINKED ISSUE #7070: Add an autofixer for no-floating-decimal
I'm using Node v6.5.0 and ESLint v3.4.0.

It would be useful to have an autofix option for the [`no-floating-decimal`](http://eslint.org/docs/rules/no-floating-decimal) rule. This would add a `0` before or after the decimal point in order to fix errors.

```js
var foo = .5;
var bar = 2.;
var baz = -.7;

// gets fixed to:

var foo = 0.5;
var bar = 2.0;
var baz = -0.7;
```

I would be willing to add this if the issue is accepted.

### PR DESCRIPTION
**Pull request by not-an-aardvark**
**What issue does this pull request address?**

#7070

**What changes did you make? (Give an overview)**

This adds an autofixer for the [`no-floating-decimal`](http://eslint.org/docs/rules/no-floating-decimal) rule.

**Is there anything you'd like reviewers to focus on?**

Nothing in particular. #7070 has not been marked as accepted yet, so this PR should not be merged before that happens.

### CODE REVIEW

**Comment by not-an-aardvark** [comment]
Not sure why CI failed; it seems like it was a timeout issue unrelated to this change.

**Comment by ilyavolodin** [comment]
Restarted CI manually. LGTM, that's was an easy fix:-)

**Comment by nzakas** [comment]
LGTM, just waiting another day for others to look.

File: lib/rules/no-floating-decimal.js [hunk 1]
```diff
@@ -27,10 +29,18 @@ module.exports = {
...
+                            message: "A leading decimal point can be confused with a dot.",
+                            fix: fixer => fixer.insertTextBefore(node, "0")
+                        });
                     }
                     if (node.raw.indexOf(".") === node.raw.length - 1) {
-                        context.report(node, "A trailing decimal point can be confused with a dot.");
+                        context.report({
+                            node,
+                            message: "A trailing decimal point can be confused with a dot.",
+                            fix: fixer => fixer.insertTextAfter(node, "0")
```
Comment (mysticatea) [inline lib/rules/no-floating-decimal.js]: Hmm. Though it's not related to this PR, this rule seems to miss `2.e+3`-like notation originally.
Was it a bug?

Comment (platinumazure) [inline lib/rules/no-floating-decimal.js]: I'm not sure it is-- the rule is specifically about decimal points being confused for dot operators, I think, and so a decimal point in the middle of a numeric literal (even if it is on the trailing each of the mantissa) should be okay.

Comment (mysticatea) [inline lib/rules/no-floating-decimal.js]: I got it, thank you for the explanation!

LGTM.

**Pull request merged**